plugins/agent-installer/worker_installer/__init__.py

```diff
@@ -50,14 +50,24 @@ def wrapper(*args, **kwargs):
         if not ctx:
             raise NonRecoverableError(
                 'CloudifyContext not found in invocation args')
-        if ctx.type == context.NODE_INSTANCE and \
-                'cloudify_agent' in ctx.node.properties:
-            agent_config = ctx.node.properties['cloudify_agent']
-        else:
-            agent_config = kwargs.get('cloudify_agent', {})
 
+        if 'cloudify_agent' in kwargs:
+            if ctx.type == context.NODE_INSTANCE and \
+                    ctx.node.properties.get('cloudify_agent'):
+                raise NonRecoverableError("'cloudify_agent' is configured "
+                                          "both as a node property and as an "
+                                          "invocation input parameter for "
+                                          "operation '{0}'"
+                                          .format(ctx.operation.name))
+            agent_config = kwargs['cloudify_agent']
+        else:
+            if ctx.type == context.NODE_INSTANCE and \
+                    ctx.node.properties.get('cloudify_agent'):
+                agent_config = ctx.node.properties['cloudify_agent']
+            else:
+                agent_config = {}
         prepare_connection_configuration(ctx, agent_config)
-
+        kwargs['cloudify_agent'] = agent_config
```

Comment:
why is this needed? 
line 76 is the one who injects the agent_config dictionary in the kwargs as 'agent_config'. 

i don't see where this assignment is used